manual/development-process/applying-or-destroying-a-bastion-host.md

```diff
@@ -0,0 +1,20 @@
+# Applying or destroying a Bastion host
+
+## Applying a Bastion host
+
+To bring up a Bastion host you must:
+
+* Go to the Bastion deploy job on [Jenkins](https://jenkins.tdr-management.nationalarchives.gov.uk/job/TDR%20Bastion%20Deploy/build)
+* From the "Stage" dropdown, choose the environment that you want to apply the bastion host to
+* From the "Command" dropdown, select "Apply"
+* **Optionally**, if you want to create an ssh tunnel you can add an ssh key.
```

Comment:
This option is less secure than using Session Manager. Is it ever necessary? Would it be better to skip it so people just follow the Session Manager instructions?

Reply:
You're still using session manager, you just go ssh-client -> session-manager -> ssh -> db-server. This is exclusively used by Phil to connect local tools to the database as using the cli in the terminal isn't useful for him. The instructions for setting up the tunnel are in the [README](https://github.com/nationalarchives/tdr-scripts) for tdr-scripts and you can see it's still using session manager underneath.
You could add something to explain all this but as I say, Phil is the only one using it and he knows what to do.

Reply:
We'll need to have to have a discussion about this with David, because using SSM manager was supposed to make the connection more secure when anyone connects, including Phil.

Reply:
I haven't tried this, but AWS do say you can do port-forwarding with Session Manager: https://aws.amazon.com/blogs/aws/new-port-forwarding-using-aws-system-manager-sessions-manager/

Does that make it possible for Phil to connect in the way he wants?

Reply:
Sorry, I think I wasn't thinking through what you said about still using session manager.

So this setup is still more secure than what we had before, because the anyone who connects to the bastion still need to use AWS SSO to make the initial connection? They won't be able to connect with _just_ an SSH key and a hostname/IP address?

Reply:
@MancunianSam Can I just check if my understanding in this last comment is correct? ☝️

Reply:
Shall we go with saying that it's a last resort to use the bastion. It should be and once there are other options to get the data from the database it will be.
For the SSH tunnel, you do need to use IAM and SSM to get to the instance first. It's in a private subnet with no open inbound ports so there's no way to get to it using traditional SSH. The port forwarding link you sent looks like a good alternative but I think it might be better to spend time creating alternative ways to get the data so we don't need the tunnel at all.